src/lib/support/BUILD.gn

```diff
@@ -145,8 +145,13 @@ static_library("support") {
     "verhoeff/Verhoeff36.cpp",
   ]
 
-  # added JNI helper on android
-  if (current_os == "android") {
+  if (current_os == "android" || getenv("JAVA_PATH") != "") {
+    import("${chip_root}/build/chip/java/rules.gni")
```

Comment:
import should be set the beginning of the build file and should use if (build_java_matter_controller) instead of getenv("JAVA_PATH") != ""

Reply:
> 2022-11-16 11:08:21 INFO    /usr/local/google/home/yufengwang/Workspace/yunhanw/connectedhomeip/out/linux-x64-java-matter-controller/../../examples/java-matter-controller/third_party/connectedhomeip/src/controller/java/CHIPDeviceController-JNI.cpp:125: undefined reference to `AndroidChipPlatformJNI_OnLoad(JavaVM_*, void*)'
> 2022-11-16 11:08:21 INFO    collect2: error: ld returned 1 exit status
> 2022-11-16 11:08:21 INFO    ninja: build stopped: subcommand failed.
> Traceback (most recent call last):
>   File "/usr/local/google/home/yufengwang/Workspace/yunhanw/connectedhomeip/./scripts/build/build_examples.py", line 195, in <module>
>     main()

As I mentioned, there is another PR which would comment out AndroidChipPlatformJNI_OnLoad

Reply:
I am not planning to add more complexity in this PR, thanks